src/app/dataset-view/additional-components/metadata-component/components/add-polling-source/add-polling-source.component.ts

```diff
@@ -0,0 +1,220 @@
+        }
+    }
+
+    public onCheckedPreprocessStep(event: Event): void {
+        const input = event.target as HTMLInputElement;
+        if (input.checked) {
+            this.pollingSourceForm.addControl(
+                SetPollingSourceSection.PREPROCESS,
+                this.fb.group({
+                    kind: this.defaultPreprocessKind,
+                }),
+            );
+        } else {
+            this.pollingSourceForm.removeControl(
+                SetPollingSourceSection.PREPROCESS,
+            );
+        }
+    }
+
+    constructor(
+        private fb: FormBuilder,
+        private createDatasetService: AppDatasetCreateService,
+        private yamlEventService: TemplatesYamlEventsService,
+        private activatedRoute: ActivatedRoute,
+        private modalService: NgbModal,
+        private cdr: ChangeDetectorRef,
+    ) {
+        super();
+    }
+
+    ngOnInit(): void {
+        this.trackSubscription(
+            this.createDatasetService.onErrorCommitEventChanges.subscribe(
+                (message: string) => {
+                    this.errorMessage = message;
+                    this.cdr.detectChanges();
+                },
+            ),
+        );
+    }
+
+    public changeStep(step: SetPollingSourceSection): void {
+        this.currentStep = step;
+    }
+
+    public onSubmit(): void {
+        this.transformSchema();
+        this.processFetchOrderControl();
+        this.trackSubscription(
+            this.createDatasetService
+                .commitEventToDataset(
+                    this.getDatasetInfoFromUrl().accountName,
+                    this.getDatasetInfoFromUrl().datasetName,
+                    this.yamlEventService.buildYamlSetPollingSourceEvent(
+                        this.pollingSourceForm.value as Omit<
+                            SetPollingSource,
+                            "__typename"
+                        >,
+                    ),
+                )
+                .subscribe(),
+        );
+    }
+
+    public getDatasetInfoFromUrl(): DatasetInfo {
+        const paramMap: ParamMap = this.activatedRoute.snapshot.paramMap;
+        return {
+            accountName: requireValue(
+                paramMap.get(ProjectLinks.URL_PARAM_ACCOUNT_NAME),
+            ),
+            datasetName: requireValue(
+                paramMap.get(ProjectLinks.URL_PARAM_DATASET_NAME),
+            ),
+        };
+    }
+
+    public onEditYaml(): void {
+        const modalRef: NgbModalRef = this.modalService.open(
+            FinalYamlModalComponent,
+            { size: "lg" },
+        );
+        this.transformSchema();
+        this.processFetchOrderControl();
+        (modalRef.componentInstance as FinalYamlModalComponent).yamlTemplate =
+            this.yamlEventService.buildYamlSetPollingSourceEvent(
+                this.pollingSourceForm.value as Omit<
+                    SetPollingSource,
+                    "__typename"
+                >,
+            );
```

Comment:
This is becoming longer, and potentially it will be even longer. Time to introduce a service, as this is not a visual work, so it should not live in the component